mozilla/bedrock
# Migrate VPN resource center to Wagtail (#14860)

## One-line summary

Migrates VPN Resource Center index and detail pages to Wagtail.

## Significant changes and points to review

One thing we might have to think about is how we plan to roll these pages out once this merges. This will be the first time we're testing localization through Smartling, and we need to be careful not to temporarily "lose" any content in the process.

Also, there were a few things in the old Contentful pages that had code but were never implemented (such as related article categories). I'm not going to port those over now, seeing as they never made it to production?

## Issue / Bugzilla link

#14860

## Testing

1. Run `make preflight`
2. Run `./manage.py makemigrations` and then `./manage.py migrate`
3. Run `npm start`.
4. Sign into the CMS: http://localhost:8000/cms-admin/
5. Publish a new Structural page called `Mozilla Products` with a slug `products`.
6. Add a child page (another Structural page) called `Mozilla VPN` with a slug `vpn`, and hit publish.
7. Navigate to the `/vpn/` structural page and add a child page of type `Vpn resource center index page`. Make sure it has a slug of `resource-center`.
8. Add a title, sub title, and a call to action snippet (replicating https://www.mozilla.org/en-US/products/vpn/resource-center/ is a good test).
9. Add a few child pages of type `Vpn resource center detail page` (this should be the only child type that's allowed by the parent).
10. Re-create some detail / article pages from https://www.mozilla.org/en-US/products/vpn/resource-center/.
11. If you create a `Call to action bottom` snippet, it should only show up when you have more than 6 child articles.


----

## Review TODOs for @stevejalim 

- [X] Check the above 
- [X] [non blocking] Raise issue about auto-translation of subtrees failing -> https://github.com/wagtail/wagtail-localize/issues/773
- [X] [non blocking]  Raise issue about translation UI https://github.com/wagtail/wagtail-localize/issues/833
- [X] Test translation of pages into more than one locale
    - [x] Fix issue where `prefer_cms` falls back to CMS-based en-US page even though other locales exist for that path outside the CMS
    - [x]  [non blocking] Test Smartling integration with real-world content 
       - [ ] https://github.com/mozilla/wagtail-localize-smartling/issues/31 
- [X] Deal with untranslated pages being Published immediately, rather than remaining drafts -> ticketed https://github.com/mozilla/bedrock/issues/15345
- [x] https://github.com/mozilla/bedrock/issues/15330
